### PR TITLE
[IA-4883] Move disk dependencies

### PR DESCRIPTION
**Pull request by jdcanas**
This PR sets the stage for moving the disk client into its own package by re-organizing dependencies/models/util functions. Specifically it:
- gets rid of `disk-models`, all models now live in a more appropriate home than a dumping ground
- makes disk client a SRP file that just performs the network calls. Utility methods that decorate the data for use in the UI are moved to the provider

### CODE REVIEW

File: src/libs/ajax/leonardo/providers/LeoDiskProvider.ts [hunk 1]
```diff
@@ -53,3 +54,75 @@ export const leoDiskProvider: LeoDiskProvider = {
...
+export interface GooglePdType {
+  value: GoogleDiskType;
+  label: string;
+  regionToPricesName: string;
+}
+
+export interface AzurePdType {
+  value: AzureDiskType;
+  label: string;
+  // TODO: Pricing skuLetter: 'S'; Enable SSD types | 'E';
```
Comment (LizBaldo) [inline src/libs/ajax/leonardo/providers/LeoDiskProvider.ts:86]: Nit, I know the purpose of this PR is to move things around to better package the disk provider, but would these TODOs be worth doing. or at least get tickets for?

File: src/libs/ajax/leonardo/models/disk-models.ts [hunk 2]
```diff
@@ -1,121 +0,0 @@
-import { Mutate } from '@terra-ui-packages/core-utils';
```
Comment (LizBaldo) [inline src/libs/ajax/leonardo/models/disk-models.ts:1]: So if I understand properly, using models for each resource is not a desirable pattern anymore, and it is better to have backend specific mapping with the ajax/disk, and everything UI facing with the provider?

**Review by LizBaldo — APPROVED** [review]
The moves make sense to me and make the flow a lot clearer, thanks! I have a couple of clarifying questions for my own understanding but by no means a blocker 👍

File: src/libs/ajax/leonardo/models/disk-models.ts (hunk 2 above)
Comment (jdcanas) [reply src/libs/ajax/leonardo/models/disk-models.ts:1]: we still want to use models for each resource, but yes backend specific mapping for the data client layer and the UI facing model in the provider (which could be the same)

File: src/libs/ajax/leonardo/providers/LeoDiskProvider.ts (hunk 1 above)
Comment (jdcanas) [reply src/libs/ajax/leonardo/providers/LeoDiskProvider.ts:86]: I think thats pending us expanding our azure disk offerings, so more of a feature enhancement than tech debt

**Review by cahrens — APPROVED** [review]
Approving changes `workspaces/common/state` files.

**Pull request merged**